packages/ui/src/components/Renderer.tsx

```diff
@@ -22,6 +22,7 @@ const Wrapper = ({
       width: schema.width * ZOOM,
       top: schema.position.y * ZOOM,
       left: schema.position.x * ZOOM,
+      transform: `rotate(${schema.rotate ?? 0}deg)`,
```

Comment:
```suggestion
      transform: `rotate(${schema.rotate ?? 0}deg)`,
      transformOrigin: 'bottom left',
```

Hi @hand-dot, 

Sorry for the slow reply.

I have just been looking at this. I think the problem is that the FE and pdflib are rotating from different positions. 
The FE was rotating from the center, but pdflib is rotating from the bottom-left corner.

I have tried setting the rotation origin as suggested here and the positioning appears to be better:


<img width="259" alt="Screenshot 2023-10-13 at 12 55 27" src="https://github.com/pdfme/pdfme/assets/7068515/011d8977-0d1d-4806-9e41-05d6c0017f0e">

<img width="340" alt="Screenshot 2023-10-13 at 12 55 59" src="https://github.com/pdfme/pdfme/assets/7068515/5b34d9da-4bc5-4975-80f2-fa521255c877">


However, there is still a positional alignment issue as you can see where the text is being aligned to the bottom of the box:

<img width="230" alt="Screenshot 2023-10-13 at 12 58 39" src="https://github.com/pdfme/pdfme/assets/7068515/63729e86-b558-45fa-ada6-d13692be274a">
<img width="330" alt="Screenshot 2023-10-13 at 12 59 10" src="https://github.com/pdfme/pdfme/assets/7068515/b6cb582e-a3ca-4d6b-894c-abe251f1e747">

I will continue to look at this and update with my findings.

Reply:
This will be quite challenging for multi-line text.
Because we draw each line separately, the point of rotation will be different for each line:

<img width="258" alt="Screenshot 2023-10-13 at 15 09 39" src="https://github.com/pdfme/pdfme/assets/7068515/7af183a0-69b9-4bde-8ffc-c88936117b5a">
<img width="334" alt="Screenshot 2023-10-13 at 15 10 26" src="https://github.com/pdfme/pdfme/assets/7068515/3b1a08e3-4399-4bbf-b627-ab6a2260f416">

@hand-dot do you remember why we draw each line separately, rather than with a single `page.drawText()` call?

Reply:
I have tried joining the lines and rendering them together in a single pdflib call, but this doesn't appear to be much better, because the rotation point is around the top line of the text, not the bottom. Here are some examples with a small rotational increment, you can see they are rotating from the bottom corner of the 'P':

<img width="338" alt="Screenshot 2023-10-13 at 17 17 58" src="https://github.com/pdfme/pdfme/assets/7068515/1152c70e-1c98-435b-8284-8fb6d2fa13b2">
<img width="343" alt="Screenshot 2023-10-13 at 17 18 22" src="https://github.com/pdfme/pdfme/assets/7068515/9405bd16-1828-43c6-b30a-7aa7f9f8e2f4">
<img width="332" alt="Screenshot 2023-10-13 at 17 20 54" src="https://github.com/pdfme/pdfme/assets/7068515/e47530a4-10b7-41a2-9e10-745b6b806fed">

I think we would need to calculate the positional offset of each row and apply them, which will be fun...

Reply:
Images and bar codes do rotate as they do in the UI though which is good:

<img width="546" alt="Screenshot 2023-10-13 at 17 23 46" src="https://github.com/pdfme/pdfme/assets/7068515/3f03e153-1ae2-4b73-a68b-57f58918f2b2">
<img width="552" alt="Screenshot 2023-10-13 at 17 23 58" src="https://github.com/pdfme/pdfme/assets/7068515/40e97c14-aca5-476d-93bd-6fd836ac3d2f">

It is only text which is going to be difficult.

Reply:
I have some ideas of how to calculate these adjustments for text schemas. I will continue trying in the next few days.

Reply:
Hey @peteward ,
Amazing!!! Thank you. I appreciate your contribution.

Reply:
> @hand-dot do you remember why we draw each line separately, rather than with a single page.drawText() call?
we are using separate page.drawText() calls for each line in order to support text alignment.